classes/Optimization/Process/AbstractProcess.php

```diff
@@ -1570,16 +1530,34 @@ public function delete_nextgen_files( $keep_full = false ) {
 	 * @since 2.2
 	 *
 	 * @param  string $file_path Path to the non-next-gen file.
-	 * @return bool|WP_Error    True on success. A \WP_Error object on failure.
+	 * @return void|WP_Error A \WP_Error object on failure.
 	 */
 	protected function delete_nextgen_file( $file_path ) {
 		if ( ! $file_path ) {
 			return new WP_Error( 'no_path', __( 'Path to non-next-gen file not provided.', 'imagify' ) );
 		}
 
+		$next_gen_ext = $this->get_format_extension( $this->format );
 		$next_gen_file = new File( $file_path );
-		$next_gen_path = $next_gen_file->get_path_to_nextgen( $this->format );
+		$this->delete_file( $next_gen_file->get_path_to_nextgen( $next_gen_ext ) );
```

Comment:
We have to handle the following cases:
- both AVIF and WebP are enabled
- only AVIF
- only WebP
- none

The design using `$this->format` is not compatible with that, because it restricts to only one next-gen formats enabled at the same time.

We should make sure all next-gen versions are deleted when this is called